src/script/page/AppLock/AppLock.tsx

```diff
@@ -131,11 +147,16 @@ const AppLock: React.FC<AppLockProps> = ({
   }, []);
 
   useEffect(() => {
+    const hasBeenEnabled = !prevAppLockEnabled.current && isAppLockEnabled;
+
     if (isAppLockEnabled) {
-      showAppLock();
-    } else if (appLockState.hasPassphrase()) {
-      appLockRepository.removeCode();
+      if (!appLockState.hasPassphrase()) {
+        appLockRepository.setDisabled();
+        return;
```

Comment:
I have added the password check here to prevent the user from being shown the applock without us having a passphrase.